src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -1734,14 +1736,34 @@ ObjectDataHandler::clone_ret ObjectDataHandler::clone(
   return with_objects_data(
     ctx,
     [ctx, this](auto &object_data, auto &d_object_data) {
+    using std::swap;
     ceph_assert(d_object_data.is_null());
     if (object_data.is_null()) {
       return clone_iertr::now();
     }
-    return ctx.tm.get_pin(ctx.t, object_data.get_reserved_data_base()
-    ).si_then([this, &object_data, &d_object_data, ctx](auto mapping) {
+    bool is_clone = ctx.onode.is_head();
```

Comment:
This is a lot more brittle than your original implementation -- requires ObjectDataHandler to guess the intent of the caller.  Let's go with your original implementation and simply add the comments.  Apologies for wasting your time on this!